Braintree/src/main/java/com/braintreepayments/api/BraintreeFragment.java

```diff
@@ -558,6 +560,10 @@ public void onActivityResult(final int requestCode, int resultCode, Intent data)
     public boolean hasFetchedPaymentMethodNonces() {
         return mHasFetchedPaymentMethodNonces;
     }
+    
+    public boolean hasMatchingAuthorization(String authorization) {
+        return mAuthorization.toString() == authorization;
```

Comment:
we probably want to do a null check here

Reply:
You perform reference check(not the actual value) between two String's. I think .equals() call should be here instead.